pkg/okteto/stream.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package okteto
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/stream"
+)
+
+var (
+	// gitDeployUrlTemplate (baseURL, namespace, dev environment name, action name)
+	gitDeployUrlTemplate = "%s/sse/logs/%s/gitdeploy/%s?action=%s"
+)
+
+type streamClient struct {
+	client *http.Client
+}
+
+func newStreamClient(httpClient *http.Client) *streamClient {
+	return &streamClient{
+		client: httpClient,
+	}
+}
+
+type pipelineLogFormat oktetoLog.JSONLogFormat
+
+// StreamPipelineLogs retrieves logs from the pipeline provided and prints them, returns error
+func (c *streamClient) PipelineLogs(ctx context.Context, name, actionName string) error {
+	streamURL := fmt.Sprintf(gitDeployUrlTemplate, Context().Name, Context().Namespace, name, actionName)
+	url, err := url.Parse(streamURL)
+	if err != nil {
+		return err
+	}
+	return stream.GetLogsFromURL(ctx, c.client, url.String(), printPipelineLog)
```

Comment:
Nit: This would make the `PipelineLogs` function hard to unit testing. It doesn't have too much logic so It might be fine, but adding it as a dependency of `streamClient` might solve it:

```
type streamer func(ctx context.Context, c *http.Client, url string, print printFn) error

....

type streamClient struct {
	client *http.Client
        streamer streamer
}

....

func newStreamClient(httpClient *http.Client) *streamClient {
	return &streamClient{
		client: httpClient,
                streamer: stream.GetLogsFromURL,
	}
}

```

Reply:
as talked offline, `PipelineLogs`  function is simple enough for the overhead that would be to do more refactor. i would leave it as is now.